app/main.py

```diff
@@ -1,54 +1,64 @@
+from app.prints import ConsolePrint, ReversePrint, Print
+from app.serializers import JsonSerializer, XmlSerializer, Serializer
 
 
 class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+    def __init__(self, title: str, content: str) -> None:
+        self.__title = title
+        self.__content = content
+
+    @property
+    def title(self) -> str:
+        return self.__title
+
+    @property
+    def content(self) -> str:
+        return self.__content
+
+    def display(self, method: Display) -> None:
+        method.display(self.content)
+
+    def print_book(self, method: Print) -> None:
+        method.print(self.title, self.content)
+
+    def serialize(self, method: Serializer) -> str:
+        return method.serialize(self.title, self.content)
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    display_types = {
+        "display_console": ConsoleDisplay(),
+        "display_reverse": ReverseDisplay(),
+        "print_console": ConsolePrint(),
+        "print_reverse": ReversePrint(),
+        "serialize_json": JsonSerializer(),
+        "serialize_xml": XmlSerializer(),
+    }
+
+    result = None
     for cmd, method_type in commands:
+        method = display_types.get(f"{cmd}_{method_type}")
+
         if cmd == "display":
-            book.display(method_type)
+            book.display(method)
         elif cmd == "print":
-            book.print_book(method_type)
+            book.print_book(method)
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            result = book.serialize(method)
```

Comment:
Put return just here or even just print, then you don't need to call print when you call main